azure-kusto-data/azure/kusto/data/_models.py

```diff
@@ -146,11 +164,17 @@ def __len__(self):
         return self.rows_count
 
     def __iter__(self):
-        for row in self.rows:
-            yield row
+        for row_index, row in enumerate(self.raw_rows):
+            if self.kusto_result_rows:
```

Comment:
You get the gist :)

Reply:
Here its not the same actually. 
In rows - we'll convert all rows, here if we already have the converted rows, use them, otherwise convert row by row